tests/worksheet_test.py

```diff
@@ -1911,4 +1912,10 @@ def test_add_validation(self):
             {"spreadsheetId": self.spreadsheet.id, "replies": [{}]},
         )
 
-        self.assertRaises(APIError, sheet.update, values="X", range_name="A1")
+        with self.assertRaises(APIError) as ex:
+            sheet.update(values="X", range_name="A1")
+
+        # Ensure that the exception is able to be pickled and unpickled
+        # Further ensure we are able to access the exception's properties after pickling
+        reloaded_exception = pickle.loads(pickle.dumps(ex.exception))  # nosec
+        self.assertEqual(reloaded_exception.args[0]["status"], "INVALID_ARGUMENT")
```

Comment:
don't you need to indent this part of the code so it's under the `with` statement, otherwise you argument `ex` is not reliably accessible after the `with` statement exists ?

Reply:
Nah, the context object `ex` remains populated after the with and the code execution actually stops inside the `with` block after the `sheet.update` as it raises the error.
See [official docs](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) for their example on this which is what I'm following.

Reply:
Alright to me.